src/components/Api.js

```diff
@@ -0,0 +1,78 @@
+class Api {
+	constructor({ baseUrl, headers }) {
+		this._baseUrl = baseUrl;
+		this._headers = headers;
+	}
+
+	_apiCall(url, method, body) {
+		return fetch(`${this._baseUrl}${url}`, {
+			method,
+			headers: this._headers,
+			body: JSON.stringify(body),
+		})
+			.then((res) => {
+				if (res.ok) {
+					return res.json();
+				}
+				return Promise.reject(`Error: ${res}`);
+			})
+			.catch((err) => {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`   after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.